### PR TITLE
Add Cellular example to CircleCI

### PR DESCRIPTION
**Pull request by noonfom**
@paul-szczepanek-arm please review

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          cd mbed-os-example-cellular
+          mbed update
+          mbed compile -t GCC_ARM -m WIO_3G || exit 1
+
+workflows:
+  version: 2
+  workflow:
+    jobs:
+    - prepare
+    - bbuild_disco_l496ag:
```
Comment (paul-szczepanek-arm) [inline .circleci/config.yml:55]: typo

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      - run:
+          working_directory: mbed-os-example-cellular
+          command: ln -s $PWD/../mbed-os mbed-os
+      - persist_to_workspace:
+          root: /root/project
+          paths: mbed-os-example-cellular
+
+  build_disco_l496ag:
+    docker:
+      - image: mbedos/mbed-os-env:stable
```
Comment (paul-szczepanek-arm) [inline .circleci/config.yml:26]: try with :latest and see if it has the correct mbed-tools (remove the `ython3 -m pip install mbed-tools --upgrade`)

Comment (noonfom) [reply .circleci/config.yml:26]: Okay. I'll do it after current CI succeeds/fails.

Comment (paul-szczepanek-arm) [reply .circleci/config.yml:26]: actually the cmake still needs that upgrade even on latest

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  build_disco_l496ag:
+    docker:
+      - image: mbedos/mbed-os-env:latest
+    working_directory: ~
+    steps:
+      - attach_workspace:
+          at: /root/project
+      - run: |
+          cd mbed-os-example-cellular
+          mbed update
```
Comment (paul-szczepanek-arm) [inline .circleci/config.yml:30]: all the mbed update no longer needed

Comment (noonfom) [reply .circleci/config.yml:30]: Got it :+1:

**Comment by noonfom** [comment]
mbed-tools fails to compile for DISCO_L496AG because there's a typo in the name of the CMake file in mbed-os/targets/TARGET_STM/TARGET_STM32L4/TARGET_STM32L496xG/TARGET_DISCO_L496AG. A patch PR needs to be opened and merged for this before the Cellular example can be added to CircleCI.

**Pull request merged**